doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_recovery.rst

```diff
@@ -0,0 +1,183 @@
+
+Further information about the default recovery firmware can be found in :ref:`suit_recovery`.
+The code for the default recovery firmware can be found in the :file:`samples/suit/recovery` directory.
+
+Updating the recovery firmware
+******************************
+
+To update the recovery firmware you can either use:
+
+* The APP_RECOVERY envelope, found in :file:`<main_application_build_directory>/build/DFU/app_recovery.suit`
+* The zip file, found in :file:`<main_application_build_directory>/build/zephyr/dfu_suit_recovery.zip``
+
+These can be used to update the recovery application the same as :file:`root.suit` or :file:`dfu_suit.zip` are used to update the main application - see :ref:`nrf54h_suit_sample` as an example.
+
+.. note::
+   The recovery application can only be updated from the main application - not when running the recovery application itself.
+
+Create custom recovery images
+*****************************
+
+To turn an application into a recovery application, the following steps have to be performed:
+
+1. For each of the recovery firmware images, ensure the following configuration is present:
+
+      .. code-block:: cfg
+
+         CONFIG_SUIT_RECOVERY=y
+         CONFIG_SUIT_MPI_GENERATE=n
+         CONFIG_SUIT_ENVELOPE_OUTPUT_MPI_MERGE=n
+         CONFIG_NRF_REGTOOL_GENERATE_UICR=n
+         CONFIG_NRF_REGTOOL_GENERATE_BICR=n
+
+#. Create the overlay files to be used by the recovery application.
+   In this guide it is assumed that for the application core they are placed in the custom recovery application directory in the :file:`boards/nrf54h20dk_nrf54h20_cpuapp.overlay` file.
+
+   The application core recovery image overlay should contain the following code:
+
+      .. code-block:: dts
+
+         / {
+            chosen {
+               zephyr,code-partition = &cpuapp_recovery_partition;
+               nrf,tz-secure-image = &cpuapp_recovery_partition;
+            };
+         };
+
+         &cpusec_cpuapp_ipc {
+            status = "okay";
+         };
+
+         &cpusec_bellboard {
+            status = "okay";
+         };
+
+   Optionally, if using the radio core recovery image, the radio core recovery image overlay should contain the following code:
+
+      .. code-block:: dts
+
+         / {
+            chosen {
+               zephyr,code-partition = &cpurad_recovery_partition;
+               nrf,tz-secure-image = &cpurad_recovery_partition;
+            };
+         };
+
+#. Add :file:`sysbuild.cmake` to the custom recovery application directory.
+   In this file add the following code:
+
+      .. code-block:: cmake
+
+         add_overlay_dts(recovery ${CMAKE_CURRENT_LIST_DIR}/boards/nrf54h20dk_nrf54h20_cpuapp.overlay)
+
+   This will ensure that when building from the main application directory the overlay file is attached to and not overwritten by the configuration comming from the main application.
```

Comment:
```suggestion
   This will ensure that when building from the main application directory the overlay file is attached to and not overwritten by the configuration coming from the main application.
```